test/support/redis_mock.rb

```diff
@@ -30,7 +30,6 @@ def start(&block)
 
     def shutdown
       @thread.kill
-      @thread.join
```

Comment:
Ok, so in the end this was the cause. Somehow the above kill doesn't reliably kill the thread, and we'd then be stuck joining forever...

Looks like a bug in JRuby to me, and I don't like that removing this join might cause use to leak threads but...